### PR TITLE
chore(website): [playground] add support for extends in eslint config

### LINKED ISSUE #5171: Website: [Playground] config should support available presets from ts-eslint and eslint
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

playground should support available presets from ts-eslint and eslint, `extends` field in eslint config

```json5
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking",
    "plugin:@typescript-eslint/strict"
```

this list should be deduced automatically from data exposed by eslint
 
### Affected URL(s)

https://typescript-eslint.io/play/

### PR DESCRIPTION
**Pull request by armano2**
## PR Checklist

- [x] Addresses an existing open issue: fixes #5171
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

> extracted from #6656 

#### "extends" field is now read while into account while parsing code

image

#### Tooltip in rules now includes url to rule doc

image

> **Warning**
> overrides do not work
> see https://github.com/typescript-eslint/typescript-eslint/blob/v6/packages/eslint-plugin/src/configs/eslint-recommended.ts
> https://github.com/eslint/eslint/blob/4c46fb3d861ca12e86f868af19778ce988238da7/lib/linter/linter.js#L1463-L1466

### CODE REVIEW

File: packages/website-eslint/src/index.js [hunk 1]
```diff
@@ -5,7 +5,8 @@ This saves us having to mock unnecessary things and reduces our bundle size.
 */
 // @ts-check
 
-import { rules } from '@typescript-eslint/eslint-plugin';
+import eslintJs from '@eslint/js';
```
Comment (armano2) [inline packages/website-eslint/src/index.js:8]: i'm not sure if this is the best way to get configs from eslint

Comment (JoshuaKGoldberg) [reply packages/website-eslint/src/index.js:8]: 🤷 it looks like it is!

File: packages/website-eslint/src/index.js [hunk 2]
```diff
@@ -24,8 +25,27 @@ exports.esquery = esquery;
...
+
+/** @type {Record<string, unknown>} */
+const configs = {};
+
+for (const name in eslintJs.configs) {
+  configs[`eslint:${name}`] = eslintJs.configs[name];
+}
+
+for (const name in plugin.configs) {
+  const value = plugin.configs[name];
```
Comment (JoshuaKGoldberg) [inline packages/website-eslint/src/index.js:42]: [Nitpick]

```suggestion
for (const [name, value] of Object.entries(plugin.configs)) {
```

I wonder if we should enable this: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-object-from-entries.md

**Review by JoshuaKGoldberg — APPROVED** [review]
Love it!!

File: packages/website-eslint/src/index.js (hunk 2 above)
Comment (armano2) [reply packages/website-eslint/src/index.js:42]: from computation point of view my code is faster 🕙 as we don't have too loop trough this 2 times :)

**Pull request merged**